### PR TITLE
[Expression] Add merge prebuilt function for merging JSON object 

### PR DESCRIPTION
**Pull request by cosmicshuai**
Parity of botbuilder-dotnet PR:
https://github.com/microsoft/botbuilder-dotnet/pull/4075

merge function can take JSON object as input.
merge(jobject1, jobject2, ...)

for example:

json1 = @"{
                   'FirstName': 'John',
                   'LastName': 'Smith',
                   'Enabled': false,
                    'Roles': [ 'User' ]
                   }"
json2 =@"{
                  'Enabled': true,
                  'Roles': [ 'User', 'Admin' ]
                 }"
string(merge(json(json1), json(json2))) will return

{"FirstName":"John","LastName":"Smith","Enabled":true,"Roles":["User","Admin"]}

### CODE REVIEW

File: libraries/adaptive-expressions/src/expressionFunctions.ts [hunk 1]
```diff
@@ -3355,7 +3355,22 @@ export class ExpressionFunctions {
...
+            new ExpressionEvaluator(ExpressionType.Merge, 
+                ExpressionFunctions.applySequenceWithError(
+                    (args: any[]): any => {
+                        let value: object = undefined;
+                        let error: string = undefined;
+                        if ((typeof(args[0]) === 'object' && !Array.isArray(args[0])) && (typeof(args[1]) === 'object' && !Array.isArray(args[1]))) {
+                            Object.assign(args[0], args[1]);
+                            value = args[0];
+                        } else {
+                            error = `The argumets ${ args[0] } and ${ args[1] } must be JSON object.`;
```
Comment (chrimc62) [inline libraries/adaptive-expressions/src/expressionFunctions.ts:3367]: >object [](start = 93, length = 6)

The arguments _ and _ must be JSON objects.

**Comment by chrimc62** [comment]
> 

To many extra lines and remove tODO

---
Refers to: libraries/adaptive-expressions/src/expressionType.ts:161 in 4a11550. [](commit_id = 4a115503a9067a3be277f23949f6f7adb0962b8c, deletion_comment = False)

File: libraries/adaptive-expressions/tests/badExpression.test.js [hunk 2]
```diff
@@ -351,6 +351,10 @@ const badExpressions =
         'jPath(hello)',// should have two params
         'jPath(hello, \'.key\')', //bad json
         'jPath(json(\'{"key1":"value1","key2":"value2"}\'), \'getTotal\')', //bad path
+        'merge(json(\'{"key1":"value1","key2":"value2"}\'))', // should have at least 2 arguments
+        'merge(json2, jarray1)', // should only have JSON object arguments
+        'merge(jarray1, json2)', // should only have JSON object arguments
+
```
Comment (chrimc62) [inline libraries/adaptive-expressions/tests/badExpression.test.js:358]: Has extra blank line.

File: libraries/adaptive-expressions/tests/expressionParser.test.js [hunk 3]
```diff
@@ -621,6 +621,8 @@ const dataSource = [
     ['coalesce(nullObj, false, \'hello\')', false],
     ['jPath(jsonStr, pathStr )', ['Jazz', 'Accord']],
     ['jPath(jsonStr, \'.automobiles[0].maker\' )', ['Nissan']],
+    ['string(merge(json1, json2))', '{"FirstName":"John","LastName":"Smith","Enabled":true,"Roles":["User","Admin"]}'],
+    ['string(merge(json1, json2, json3))', '{"FirstName":"John","LastName":"Smith","Enabled":true,"Roles":["User","Admin"],"age":36}'],
```
Comment (chrimc62) [inline libraries/adaptive-expressions/tests/expressionParser.test.js:625]: >"Roles":["User","Admin"] [](start = 98, length = 24)

Is this that last definition wins?  Or is this actually merging elements in an array?

File: libraries/adaptive-expressions/src/expressionFunctions.ts (hunk 1 above)
Comment (cosmicshuai) [reply libraries/adaptive-expressions/src/expressionFunctions.ts]: fixed

File: libraries/adaptive-expressions/tests/badExpression.test.js (hunk 2 above)
Comment (cosmicshuai) [reply libraries/adaptive-expressions/tests/badExpression.test.js]: removed the extra blank line

File: libraries/adaptive-expressions/tests/expressionParser.test.js (hunk 3 above)
Comment (cosmicshuai) [reply libraries/adaptive-expressions/tests/expressionParser.test.js]: I changed the test case to make the more clear on merging arrays. The property in the latter input will replace the same property in the previous JSON object.

**Pull request merged**